xcp_d/utils/modified_data.py

```diff
@@ -137,28 +137,32 @@ def _drop_dummy_scans(bold_file, dummy_scans):
     """
     # read the bold file
     bold_image = nb.load(bold_file)
+    ndim = bold_image.ndim
```

Comment:
The changes to _drop_dummy_scans may be unnecessary. `Nifti1Image.slicer[]` seems to use more memory than loading the data as a numpy array, slicing it, and constructing a new image, but I tried it out in this function and then reverted it. Hence the changes.